File: atcoder/abc413/src/bin/a.py
di = [0, 1, 0, -1, 1, 1, -1, -1]
dj = [1, 0, -1, 0, 1, -1, 1, -1]

n = int(input())
s = [input() for _ in range(n)]

ss = set()
for i in range(n):
    for j in  range(n):
        if i==j: continue
        t = s[i] + s[j]
        ss.add(t)
print(len(ss))

File: atcoder/abc413/src/bin/a.rs
#[allow(unused_imports)]
use proconio::input;

fn main() {
    input!{
        n: usize,
        m: u64,
        a: [u64; n],
    }

    let s = a.iter().sum::<u64>();
    println!("{}", if s<=m { "Yes" } else { "No" });
}
